resources/lib/play_utils.py

```diff
@@ -840,12 +841,12 @@ def audio_subs_pref(url, list_item, media_source, item_id, audio_stream_index, s
         else:  # User backed out of selection
             playurlprefs += "&AudioStreamIndex=%s" % default_audio
 
-    # set subtitle index
+    # Set subtitle index
     if select_subs_index is not None:
         # Load subtitles in the listitem if downloadable
         if select_subs_index in downloadable_streams:
-            subtitle_url = "%s/Videos/%s/%s/Subtitles/%s/Stream.srt"
-            subtitle_url = subtitle_url % (settings.getSetting('server_address'), item_id, source_id, select_subs_index)
+            delivery_url = downloadable_streams[select_subs_index]['DeliveryUrl']
```

Comment:
```suggestion
            delivery_url = media_streams[select_subs_index]['DeliveryUrl']
```
I suspect this is what you were going for.  `downloadable_streams` is a list of ints, not a list of dicts